test/explorer/series/duration_test.exs

```diff
@@ -284,6 +463,116 @@ defmodule Explorer.Series.DurationTest do
+      assert Series.to_list(product_s) == [two_hour_duration_s]
+    end
+
+    test "Integer * duration[μs]" do
+      one_hour_us_s = Series.from_list([@one_hour_duration_us])
+      product_s = Series.multiply(2, one_hour_us_s)
+
+      assert product_s.dtype == {:duration, :microsecond}
+      two_hour_duration_s = %Duration{value: 2 * @one_hour_us, precision: :microsecond}
+      assert Series.to_list(product_s) == [two_hour_duration_s]
+    end
+
+    test "duration[μs] * Integer" do
+      one_hour_us_s = Series.from_list([@one_hour_duration_us])
+      product_s = Series.multiply(one_hour_us_s, 2)
+
+      assert product_s.dtype == {:duration, :microsecond}
+      two_hour_duration_s = %Duration{value: 2 * @one_hour_us, precision: :microsecond}
+      assert Series.to_list(product_s) == [two_hour_duration_s]
+    end
+
+    # Float
+
+    test "float * duration[μs]" do
+      one_hour_us_s = Series.from_list([@one_hour_duration_us])
+      two_s = Series.from_list([2.0])
+      product_s = Series.multiply(two_s, one_hour_us_s)
+
+      assert product_s.dtype == {:duration, :microsecond}
+      two_hour_duration_s = %Duration{value: 2 * @one_hour_us, precision: :microsecond}
+      assert Series.to_list(product_s) == [two_hour_duration_s]
+    end
+
+    test "duration[μs] * float" do
+      one_hour_us_s = Series.from_list([@one_hour_duration_us])
+      two_s = Series.from_list([2.0])
+      product_s = Series.multiply(one_hour_us_s, two_s)
+
+      assert product_s.dtype == {:duration, :microsecond}
+      two_hour_duration_s = %Duration{value: 2 * @one_hour_us, precision: :microsecond}
+      assert Series.to_list(product_s) == [two_hour_duration_s]
+    end
+
+    test "Float * duration[μs]" do
+      one_hour_us_s = Series.from_list([@one_hour_duration_us])
+      product_s = Series.multiply(2.0, one_hour_us_s)
+
+      assert product_s.dtype == {:duration, :microsecond}
+      two_hour_duration_s = %Duration{value: 2 * @one_hour_us, precision: :microsecond}
+      assert Series.to_list(product_s) == [two_hour_duration_s]
+    end
+
+    test "duration[μs] * Float" do
+      one_hour_us_s = Series.from_list([@one_hour_duration_us])
+      product_s = Series.multiply(one_hour_us_s, 2.0)
+
+      assert product_s.dtype == {:duration, :microsecond}
+      two_hour_duration_s = %Duration{value: 2 * @one_hour_us, precision: :microsecond}
+      assert Series.to_list(product_s) == [two_hour_duration_s]
+    end
+
+    test "fractional parts of floats work (roughly) as expected" do
+      # This test is not exhaustive. Rather, its purpose is to give us a reasonable confidence that
+      # multiplying durations by floats is fairly accurate.
+      #
+      # The exact answers we see here are subject to implementation details outside our control.
+      # If we find that this test breaks unexpectedly (e.g. from a dependency update), then we may
+      # wish to remove it.
```

Comment:
I included this test mostly because I wanted to see what we could expect from `float * duration`. It is quite brittle and I'm happy to remove it.